### PR TITLE
Add Ibex_AXI submodule

### PR DESCRIPTION
**Pull request by choucl**

### CODE REVIEW

**Comment by yuyuranium** [comment]
Please provide the `fusesoc` command to compile the `ibex_axi` core into Vivado project in this repository. Thanks!

**Comment by choucl** [comment]
The usage of `fusesoc` has been append to the readme of the submodule. Please take a look.

**Comment by yuyuranium** [comment]
Should I `cd` into `pynq_z2/pl` first before running `fusesoc`?

**Comment by choucl** [comment]
I think it based on where you add and install the fusesoc libraries. If you add and install the fusesoc libraries inside the `pl` directory, the `fusesoc.conf` will be generated there, then you should run the `fusesoc run` inside `pl` so that it could read the configuration of the libraries.

**Comment by yuyuranium** [comment]
Got it, thanks! Would you provide the instruction for a fresh clone to build the Vivado project? You can edit `README.md` of this repo.

**Comment by choucl** [comment]
No problem. The step instructions has been added to the REAME file.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,29 @@
...
+### Steps
+- Install fusesoc
+    ```
+    $ pip3 install fusesoc==2.1
+    ```
+- Install git libraries
+    ```
+    $ fusesoc library add --sync-type git ibex https://github.com/lowRISC/ibex.git`          # ibex library
+    $ fusesoc library add --sync-type git pulp-axi https://github.com/pulp-platform/axi.git` # pulp-axi library
+    $ fusesoc library add --sync-type local ibex-axi [relate path to pynq/pl]`               # ibex-axi local library
```
Comment (yuyuranium) [inline README.md]:
```suggestion
    $ fusesoc library add --sync-type git ibex https://github.com/lowRISC/ibex.git          # ibex library
    $ fusesoc library add --sync-type git pulp-axi https://github.com/pulp-platform/axi.git # pulp-axi library
    $ fusesoc library add --sync-type local ibex-axi [relate path to pynq/pl]               # ibex-axi local library
```
Are these glitches?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
+    ```
+- Install ibex dependencies 
+    ```
+    $ cd fusesoc_libraries/ibex
+    $ pip3 install -r python-requirements.txt
+    ```
+- Run and build Vivado project
+    ```
+    $ fusesoc run caslab:rtos:ibex_axi:1.0.0 --pnr none --jobs [JOBS]
```
Comment (yuyuranium) [inline README.md]: Can I run this command in the repository root?

**Review by yuyuranium — COMMENTED** [review]
Great, thank you! Would you also add the `build` directory to `.gitignore` so that the repo won't be poluted?

File: README.md (hunk 2 above)
Comment (choucl) [reply README.md]: You should run this at where the `fusesoc.conf` lives. It should be the place you run `fusesoc library add`.

**Comment by yuyuranium** [comment]
Great! Thank you! :+1:

**Review by yuyuranium — APPROVED** [review]

**Pull request merged**